### PR TITLE
✨ Update SignedReleases to use RepoClient API

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Use RepoClient API in `Signed-Releases` check.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

File: clients/githubrepo/graphql.go [hunk 1]
```diff
@@ -152,6 +173,24 @@ func pullRequestFrom(data *graphqlData) []clients.PullRequest {
 	return ret
 }
 
+func releasesFrom(data *graphqlData) []clients.Release {
+	releases := make([]clients.Release, 0)
```
Comment (oliverchang) [inline clients/githubrepo/graphql.go:177]: nit: do we need the ", 0" (or the make() at all for that matter)? https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices also says to prefer `var releases []clients.Release`

File: clients/githubrepo/graphql.go [hunk 2]
```diff
@@ -152,6 +173,24 @@ func pullRequestFrom(data *graphqlData) []clients.PullRequest {
...
+	for _, r := range data.Repository.Releases.Nodes {
+		release := clients.Release{
+			TagName: string(r.TagName),
+			Assets:  make([]clients.ReleaseAsset, 0),
```
Comment (oliverchang) [inline clients/githubrepo/graphql.go:181]: nit: no need for this. you can append to nil arrays (unless you want to preallocate the capacity here with `len(r.ReleaseAssets.Nodes)`

**Review by oliverchang — APPROVED** [review]

File: clients/githubrepo/graphql.go (hunk 1 above)
Comment (azeemshaikh38) [reply clients/githubrepo/graphql.go:177]: Done.

File: clients/githubrepo/graphql.go (hunk 2 above)
Comment (azeemshaikh38) [reply clients/githubrepo/graphql.go:181]: Done.

**Pull request merged**